tests/checks/default-setup-path.fish

```diff
@@ -0,0 +1,12 @@
+#RUN: %fish -C 'set -g fish %fish' %s
+
+# FIXME: Rationalize behavior when PATH is explicitly unset, should this not behave like PATH=""?
+# "" is threated like ".", see https://github.com/fish-shell/fish-shell/issues/3914
+if command -q getconf
+    env -u PATH $fish -c 'test "$PATH" = "$('(command -s getconf)' PATH)"; and echo Success'
+else
+    # this is DEFAULT_PATH
+    # This techinically depends on PREFIX set in CMake
+    env -u PATH $fish -c 'test "$PATH" = "/usr/local/bin:/usr/bin:/bin"; and echo Success'
```

Comment:
I think we should make this test work with any prefix, for example by testing only `$PATH[2 3]`

Reply:
I used `[2..]` to ensure it checks all elements